enzyme/Enzyme/Enzyme.cpp

```diff
@@ -79,6 +79,220 @@ llvm::cl::opt<bool>
 
 namespace {
 
+static void
+handleCustomDerivative(StringRef handlername, int numargs, llvm::Module &M,
+                       llvm::GlobalVariable &g,
+                       std::vector<GlobalVariable *> &globalsToErase) {
+  if (g.hasInitializer()) {
+    if (auto CA = dyn_cast<ConstantAggregate>(g.getInitializer())) {
+      if (CA->getNumOperands() != numargs) {
+        llvm::errs() << M << "\n";
+        llvm::errs() << "Use of " << handlername
```

Comment:
This is a good start, though I still think making (maybe a template on the number of args) that extracts out the initializer/constantexpr code from this function and below would be really useful.